### PR TITLE
Update chaincode integration tool

### PR DESCRIPTION
**Pull request by jt-nti**
Add basic CCAAS tests and improve test isolation with separate channels

Signed-off-by: James Taylor <jamest@uk.ibm.com>

### CODE REVIEW

**Comment by denyeart** [comment]
@mbwhite Could you review this one?

File: tools/chaincode-integration/src/step-definitions/channel/channel.ts [hunk 1]
```diff
@@ -39,14 +39,19 @@ export class Channel {
         await this.workspace.infrastructureProvider.deployCC(this.workspace, channelName, chaincodeName);
     }
 
-    @given(/Chaincode ['"](.*)['"] has been started in server mode$/, '', TIMEOUT)
-    public async startCCAAS(chaincodeName: string): Promise<void> {
-        await this.workspace.infrastructureProvider.startCCAAS(this.workspace, chaincodeName);
-    }
+    // TODO delete me?
```
Comment (mbwhite) [inline tools/chaincode-integration/src/step-definitions/channel/channel.ts]: yes...

File: tools/chaincode-integration/src/step-definitions/channel/channel.ts [hunk 2]
```diff
@@ -39,14 +39,19 @@ export class Channel {
...
+    //     await this.workspace.infrastructureProvider.startCCAAS(this.workspace, chaincodeName);
+    // }
 
     @given(/All peers on channel ['"](.*)['"] have deployed the chaincode-as-a-service ['"](.*)['"]$/, '', TIMEOUT)
     public async deployCCAAS(channelName: string, chaincodeName: string): Promise<void> {
-        await this.workspace.infrastructureProvider.deployCC(this.workspace, channelName, `${chaincodeName}-ccaas`);
+        // Removed "-ccaas" suffix because it causes problems with finding the chaincode config
+        //   TypeError: Cannot read properties of undefined (reading 'path')
+        //   at TestNetworkProvider.deployCCAAS (/Users/jamest/Code/fabric-test/tools/chaincode-integration/src/infrastructure/testNetworkProvider.ts:89:96)
+        // TODO are there any implications? i.e. same chaincode name as a service and as normal in same test run?
```
Comment (mbwhite) [inline tools/chaincode-integration/src/step-definitions/channel/channel.ts]: won't there be a collision in the peer if there's a chaincode of the same name?

**Comment by jt-nti** [comment]
@denyeart still a work in progress but I'll nag him when it's ready :)

File: tools/chaincode-integration/src/step-definitions/chaincode/chaincode.ts [hunk 3]
```diff
@@ -54,6 +54,7 @@ export class Chaincode {
         this.workspace.actingChaincode = chaincode;
     }
 
+    // TODO delete me? Chaincode is just chaincode?
```
Comment (mbwhite) [inline tools/chaincode-integration/src/step-definitions/chaincode/chaincode.ts]: thinking this would b ejust be there incase anything setupy was needed

**Review by mbwhite — COMMENTED** [review]
SFSG 👍

File: tools/chaincode-integration/src/step-definitions/channel/channel.ts (hunk 2 above)
Comment (jt-nti) [reply tools/chaincode-integration/src/step-definitions/channel/channel.ts]: I was hoping that different features would start with a clean slate, but apparently not

**Review by mbwhite — APPROVED** [review]

**Pull request merged**